### ISSUE TITLE
"Condition not met" errors

### ISSUE THREAD

**Issue by lencioni** [microsoft/deoptexplorer-vscode#23]
Thank you for building and releasing this tool!

I am attempting to load a log file into Deopt Explorer, but unfortunately it is not yet working for me.

Running node with these flags (copied from https://devblogs.microsoft.com/typescript/introducing-deopt-explorer/?ref=refind#generating-a-log):

```sh
--log-deopt \
--redirect-code-traces \
--redirect-code-traces-to=\\.\NUL \
--log-ic \
--log-maps \
--log-maps-details \
--log-code \
--log-source-code \
--prof \
--log-internal-timer-events \
--detailed-line-info \
--logfile=v8.log \
--no-logfile-per-isolate \
--unhandled-rejections=strict
```

This is the first line logged:

```
[INFO: 16:29:09.153] Detected log for V8 9.3.345.19-node.14
```

It tries to load the file for a while, and I see a bunch of errors (hundreds or maybe thousands) like the following logged in the Output:

```
[ERR!: 16:29:09.535] line 3411: Condition not met
Error: Condition not met
    at t.assert (/Users/joe_lencioni/.vscode/extensions/rbuckton.deoptexplorer-vscode-1.1.0-darwin-arm64/dist/extension.js:2:80962)
    at t.LogProcessor.processMapEvent (/Users/joe_lencioni/.vscode/extensions/rbuckton.deoptexplorer-vscode-1.1.0-darwin-arm64/dist/extension.js:2:186804)
    at u.dispatchLogRow_ (/Users/joe_lencioni/.vscode/extensions/rbuckton.deoptexplorer-vscode-1.1.0-darwin-arm64/dist/extension.js:165:35867)
    at u.processLogLine_ (/Users/joe_lencioni/.vscode/extensions/rbuckton.deoptexplorer-vscode-1.1.0-darwin-arm64/dist/extension.js:165:36324)
    at u.processLogLine (/Users/joe_lencioni/.vscode/extensions/rbuckton.deoptexplorer-vscode-1.1.0-darwin-arm64/dist/extension.js:165:34613)
    at t.LogProcessor.processContent (/Users/joe_lencioni/.vscode/extensions/rbuckton.deoptexplorer-vscode-1.1.0-darwin-arm64/dist/extension.js:2:174533)
    at t.measureAsync (/Users/joe_lencioni/.vscode/extensions/rbuckton.deoptexplorer-vscode-1.1.0-darwin-arm64/dist/extension.js:5:27435)
    at t.LogProcessor.process (/Users/joe_lencioni/.vscode/extensions/rbuckton.deoptexplorer-vscode-1.1.0-darwin-arm64/dist/extension.js:2:174807)
    at /Users/joe_lencioni/.vscode/extensions/rbuckton.deoptexplorer-vscode-1.1.0-darwin-arm64/dist/extension.js:5:41968
```

I also spotted this in the logs:

```
[ERR!: 16:39:25.422] dropping: overflow
```

After waiting a bit more, the extension usually crashes and the logs are cleared. I managed to have it run once where instead of crashing it just kept running, and I saw the following logged:

```
[INFO: 16:40:43.129] processContent took 83,184.0ms
[INFO: 16:40:43.184] Error: ENOENT: no such file or directory, realpath '/Users/joe_lencioni/repo/node_modules/ts-node/src/index.ts'
    at Function.realpathSync.native (node:fs:2602:3)
    at Function.native (node:electron/js2c/asar_bundle:2:4804)
    at h (/Users/joe_lencioni/.vscode/extensions/rbuckton.deoptexplorer-vscode-1.1.0-darwin-arm64/dist/extension.js:5:28829)
    at f (/Users/joe_lencioni/.vscode/extensions/rbuckton.deoptexplorer-vscode-1.1.0-darwin-arm64/dist/extension.js:5:28959)
    at /Users/joe_lencioni/.vscode/extensions/rbuckton.deoptexplorer-vscode-1.1.0-darwin-arm64/dist/extension.js:2:110463
    at Array.map (<anonymous>)
    at t.SourceMap.get sources [as sources] (/Users/joe_lencioni/.vscode/extensions/rbuckton.deoptexplorer-vscode-1.1.0-darwin-arm64/dist/extension.js:2:110434)
    at /Users/joe_lencioni/.vscode/extensions/rbuckton.deoptexplorer-vscode-1.1.0-darwin-arm64/dist/extension.js:2:175455
    at t.measureAsync (/Users/joe_lencioni/.vscode/extensions/rbuckton.deoptexplorer-vscode-1.1.0-darwin-arm64/dist/extension.js:5:27435)
    at t.LogProcessor.process (/Users/joe_lencioni/.vscode/extensions/rbuckton.deoptexplorer-vscode-1.1.0-darwin-arm64/dist/extension.js:2:174925)
    at /Users/joe_lencioni/.vscode/extensions/rbuckton.deoptexplorer-vscode-1.1.0-darwin-arm64/dist/extension.js:5:41968 {
  errno: -2,
  syscall: 'realpath',
  code: 'ENOENT',
  path: '/Users/joe_lencioni/repo/node_modules/ts-node/src/index.ts'
}
```

Unfortunately, I cannot share the code that I am running this on, but I can say that I am running this code using `ts-node/register` with the swc transpiler, in case that matters here.

I also am running with some code like the following, which overrides node's `Module._resolveFilename` to enable some special import path formats:

```ts
const Module = require('module');

const transformPath = require('.');

const origResolveFilename = Module._resolveFilename;

Module._resolveFilename = function _resolveFilename(request, ...extraArgs) {
  return origResolveFilename(transformPath(request), ...extraArgs);
}
```

Please let me know if there is some more info I can provide.

**Comment by rbuckton** [comment]
The v1.1.1 version that I just published should help with the ENOENT error you saw in the log.

I have an idea as to the possible cause of the "Condition not met" error, and I'm looking into a fix. In this case, it's coming from a `map` transition that is pointing to a prior `map` that wasn't recorded or couldn't be resolved.

**Comment by rbuckton** [comment]
I've made a minor change in #28 related to the "Condition not met" error. I've switched this to instead log a warning to the output channel with a better description, and to try and use whatever information is available when recording the update. However, I don't think this change will have a major impact on the crashes you were seeing. In the meantime, I'd suggest trying again with the latest version of the extension to check if you are still seeing a crash.

I don't have a repro for the "condition not met" issue at present. Without a repro this is likely the best I can do to resolve this.
